client/order/epochqueue.go

```diff
@@ -37,13 +39,23 @@ func NewEpochQueue() *EpochQueue {
 
 // Reset clears the epoch queue. This should be called when a new epoch begins.
 func (eq *EpochQueue) Reset() {
+	atomic.StoreUint64(&eq.epoch, 0)
```

Comment:
Depending on when `Reset` is called, an atomic load/store may not be sufficient.  Note the possible sequence:
1. `Enqueue` loads `eq.epoch` and starts building the `*epochOrder`
2. `Reset` changes `eq.epoch` and `eq.orders` while `Enqueue` is running.
3. `Enqueue` possibly updates `eq.epoch` if `eq.epoch` was 0 when `Enqueue` started.
4. `Enqueue` stores the `*epochOrder` in the `eq.orders` but it's not the same epoch as when Enqueue started.